ospf6d/ospf6_network.h

```diff
@@ -36,4 +36,16 @@ extern int ospf6_recvmsg(struct in6_addr *src, struct in6_addr *dst,
 			 ifindex_t *ifindex, struct iovec *message,
 			 int ospf6_sock);
 
+#define OSPF6_MESSAGE_WRITE_ON(oi)                                             \
+	do {                                                                   \
+		if ((oi)->on_write_q == 0) {                                   \
+			listnode_add(oi->area->ospf6->oi_write_q, (oi));       \
+			(oi)->on_write_q = 1;                                  \
+		}                                                              \
+		if (!list_isempty(oi->area->ospf6->oi_write_q))                \
```

Comment:
IMO, thread_add_write() should be called if and only if the list was previously empty and now its not.

Reply:
Yeah I think you're right. Probably should change this in ospfd too.